src/cljs/rems/administration/create_catalogue_item.cljs

```diff
@@ -174,37 +190,63 @@
 (rf/reg-fx
  ::fetch-catalogue-item
  (fn [id]
-   (when id (fetch-catalogue-item id))))
+   (fetch-catalogue-item id)))
+
+(defn- item-by-id [items id-key id]
+  (first (filter #(= (id-key %) id) items)))
 
 (rf/reg-event-db
  ::fetch-catalogue-item-result
  (fn [db [_ {:keys [wfid resource-id formid localizations organization]}]]
    (-> db
-       (assoc ::form {:workflow-id wfid
-                      :resource-id resource-id
-                      :form-id formid
+       (assoc ::form {:workflow (item-by-id (::workflows db) :id wfid)
+                      :resource (item-by-id (::resources db) :id resource-id)
+                      :form (item-by-id (::forms db) :form/id formid)
                       :organization organization
                       :title (map-vals :title localizations)
                       :infourl (map-vals :infourl localizations)})
-       (dissoc ::loading-catalogue-item?))))
+       (dissoc ::loading-catalogue-item?
+               ::loading?))))
 
 ;;;; UI
 
 (def ^:private context
   {:get-form ::form
    :update-form ::set-form-field})
 
+(def ^:private organization-dropdown-id "organization-dropdown")
 (def ^:private workflow-dropdown-id "workflow-dropdown")
 (def ^:private resource-dropdown-id "resource-dropdown")
 (def ^:private form-dropdown-id "form-dropdown")
 
+(defn- list-organizations []
+  (let [workflows @(rf/subscribe [::workflows])
+        resources @(rf/subscribe [::resources])
+        forms @(rf/subscribe [::forms])
+        items-to-organizations (fn [items key]
+                                 (set (distinct (mapv key items))))]
+    (sort
+     (vec (set/intersection (items-to-organizations workflows :organization)
+                            (items-to-organizations resources :organization)
+                            (items-to-organizations forms :form/organization))))))
+
 (defn- catalogue-item-organization-field []
-  ;; TODO: Allow editing the organization of an existing catalogue item?
-  (let [readonly (or @(rf/subscribe [::editing?])
+  (let [organizations (list-organizations)
```

Comment:
does the code now assume that you pick the organization first? what if you try to pick the resource first? maybe we can leave this for future development

Reply:
As discussed, I'll change it so that the field is disabled if organization has not been selected.